skyportal/handlers/api/source.py

```diff
@@ -732,6 +733,18 @@ class Validator(Schema):
                     .filter(Comment.obj_id == obj_id)
                     .all()
                 )
+                comments_on_spectra = (
+                    CommentOnSpectrum.query_records_accessible_by(
+                        self.current_user,
+                        options=[
+                            joinedload(CommentOnSpectrum.author),
+                            joinedload(CommentOnSpectrum.groups),
+                        ],
+                    )
+                    .filter(CommentOnSpectrum.obj_id == obj_id)
+                    .all()
+                )
+                comments = comments + comments_on_spectra
```

Comment:
I'd do something like this (esp. in case we add more object-related commentable entities in the future):

```suggestion
                comments = []
                for comment_type in (Comment, CommentOnSpectrum):
                    comments.extend(
                        comment_type.query_records_accessible_by(
                            self.current_user,
                            options=[
                                joinedload(comment_type.author),
                                joinedload(comment_type.groups),
                            ],
                        )
                        .filter(comment_type.obj_id == obj_id)
                        .all()
                    )
```

Reply:
@stefanv would you agree?